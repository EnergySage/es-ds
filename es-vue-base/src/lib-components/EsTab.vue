<template>
    <b-tab
        class="mt-3"
        :title="title">
        <slot />
    </b-tab>
</template>

<script lang="js">
import { BTab } from 'bootstrap-vue';

export default {
    name: 'EsTab',
    components: {
        BTab,
    },
    props: {
        /**
         * title
         */
        title: {
            type: String,
            default: 'Title',
        },
    },
};
</script>
