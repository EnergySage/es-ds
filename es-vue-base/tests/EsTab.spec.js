import { mount } from '@vue/test-utils';
import EsTab from '@/src/lib-components/molecules/EsTab.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsTab', () => {
    // Basic test; should exist for most components
    test('<EsTab />', async () => {
        const wrapper = mount(EsTab, {
            ...jestVue,
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });

    // Test props are there
    test('<EsTab title="Title"/>', async () => {
        const wrapper = mount(EsTab, {
            ...jestVue,
            propsData: {
                title: 'Title',
            },
        });

        expect(wrapper.props('title')).toBe('Title');
        expect(wrapper.html()).toMatchSnapshot();
    });

    // Test slot is there
    test('<EsTab>My Content</EsTab>', async () => {
        const wrapper = mount(EsTab, {
            ...jestVue,
            slots: {
                default: 'My Content',
            },
        });

        expect(wrapper.vm.$slots.default[0].text).toBe('My Content');
        expect(wrapper.html()).toMatchSnapshot();
    });
});
