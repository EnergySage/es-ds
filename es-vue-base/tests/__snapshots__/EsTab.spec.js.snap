// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsTab <EsTab /> 1`] = `
"<transition-stub name="" css="true" mode="out-in" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="" leaveactiveclass="" class="mt-100">
  <div role="tabpanel" aria-hidden="true" class="tab-pane" style="display: none;" id="__BVID__4" aria-labelledby="__BVID__4___BV_tab_button__"></div>
</transition-stub>"
`;

exports[`EsTab <EsTab title="Title"/> 1`] = `
"<transition-stub name="" css="true" mode="out-in" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="" leaveactiveclass="" class="mt-100">
  <div role="tabpanel" aria-hidden="true" class="tab-pane" style="display: none;"></div>
</transition-stub>"
`;

exports[`EsTab <EsTab>My Content</EsTab> 1`] = `
"<transition-stub name="" css="true" mode="out-in" enterclass="" leaveclass="show" entertoclass="show" leavetoclass="" enteractiveclass="" leaveactiveclass="" class="mt-100">
  <div role="tabpanel" aria-hidden="true" class="tab-pane" style="display: none;">My Content</div>
</transition-stub>"
`;
